### ISSUE TITLE
[Contributors Wanted] Implement MaxPool2D

### ISSUE THREAD

**Issue by hazemessamm** [hazemessamm/hera#1]
Implement Max Pooling 2D using JAX

`__init__` should signature the following signature:

`def __init__(pool_size, stride, padding):`

Steps:
- Open a seperate branch for it.
- Implement a function that does max pooling in `functional.py`.
- Implement a `MaxPool2D` in `pooling` directory, file name: `max_pooling_2d.py`.
- Implement max pooling using `jax.lax.reduce_window`.
-  Max pooling example: lax.reduce_window(inputs, -jnp.inf, lax.max, pool_size, strides, padding).
- The class should inheret from `hera.nn.modules.module.Module` class.
- Implement a test file called `max_pooling_2d_tests.py` in `pooling` directory.
